### PR TITLE
Feature/71 button

### PR DESCRIPTION
**Pull request by ElishaSamPeterPrabhu**
Add Clickable Property
Add handlers for touch action button all platforms

### CODE REVIEW

File: Trimble.Modus.Components/Controls/Button/CustomButton.xaml.cs [hunk 1]
```diff
@@ -110,174 +122,234 @@ public new Color BackgroundColor
...
+                customButton.buttonLabel.TextColor = ResourcesDictionary.ColorsDictionary("White");
+                customButton.tempColor = customButton.buttonFrame.BackgroundColor;
+
+                break;
+
+            case ButtonColor.Tertiary:
+                customButton.buttonFrame.BackgroundColor = ResourcesDictionary.ColorsDictionary("TertiaryButton");
+                customButton.buttonFrame.Stroke = ResourcesDictionary.ColorsDictionary("Transparent");
+                customButton.buttonLabel.TextColor = ResourcesDictionary.ColorsDictionary("TrimbleGray");
+                customButton.tempColor = customButton.buttonFrame.BackgroundColor;
```
Comment (Sofiya-kumar) [inline Trimble.Modus.Components/Controls/Button/CustomButton.xaml.cs:256]: Why templcolor required?

**Review by Sofiya-kumar — CHANGES_REQUESTED** [review]

File: Trimble.Modus.Components/Controls/Button/CustomButton.xaml.cs [hunk 2]
```diff
@@ -110,174 +122,234 @@ public new Color BackgroundColor
...
-                break;
-        }
-    }
+                break;
+            default:
+                break;
+        }
+    }
+
+    private Color getOnClickColor(Color color)
```
Comment (Sofiya-kumar) [inline Trimble.Modus.Components/Controls/Button/CustomButton.xaml.cs:293]: Code refactoring needed

**Review by Sofiya-kumar — CHANGES_REQUESTED** [review]

File: Trimble.Modus.Components/Controls/Button/CustomButton.xaml.cs (hunk 1 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/Button/CustomButton.xaml.cs:256]: Its required to set the color of that particular style (fill,outline,borderless) so after setting the color of the background while clicking it ,we are s betting the color back to the previous color of the background

File: Trimble.Modus.Components/Controls/Button/CustomButton.xaml.cs (hunk 2 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/Button/CustomButton.xaml.cs:293]: Formatted

File: Trimble.Modus.Components/Controls/Button/CustomButton.xaml.cs (hunk 1 above)
Comment (rthanga1) [reply Trimble.Modus.Components/Controls/Button/CustomButton.xaml.cs:256]: Can you rename the variable? like 'ActiveColor'?

File: Trimble.Modus.Components/Handlers/TMButtoniOSTouchHandler.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+using Trimble.Modus.Components.Controls.Button;
+using UIKit;
+
+namespace Trimble.Modus.Components.Handlers;
+
+public class TMButtoniOSTouchHandler : VisualElementRenderer<CustomButton>
+{
+    public TMButtoniOSTouchHandler()
+    {
+        if (Element.IsDisabled)
```
Comment (rthanga1) [inline Trimble.Modus.Components/Handlers/TMButtoniOSTouchHandler.cs:13]: The Element is null at the constructor. We have to use OnElementChanged override method to do this

**Review by rthanga1 — CHANGES_REQUESTED** [review]
Nothing is shown in iOS.

simulator_screenshot_84E6B58D-698E-4E16-8AAE-A6B216BE19DF

File: Trimble.Modus.Components/Controls/Button/CustomButton.xaml.cs (hunk 1 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/Button/CustomButton.xaml.cs:256]: Sure

File: Trimble.Modus.Components/Handlers/TMButtoniOSTouchHandler.cs (hunk 3 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Handlers/TMButtoniOSTouchHandler.cs:13]: Modified the isdisabled handling by moving it on action occurrence

File: Trimble.Modus.Components/Controls/Button/CustomButton.xaml.cs [hunk 4]
```diff
@@ -110,174 +122,235 @@ public new Color BackgroundColor
...
+            CheckButtonStyle(customButton);
+        }
+    }
+
+    private static void CheckButtonStyle(CustomButton customButton)
+    {
+        switch (customButton.ButtonStyle)
+        {
+            case Enums.ButtonStyle.BorderLess:
+                customButton.buttonFrame.BackgroundColor = ResourcesDictionary.ColorsDictionary("Transparent");
```
Comment (Sofiya-kumar) [inline Trimble.Modus.Components/Controls/Button/CustomButton.xaml.cs:179]: Can you replace the strings with colors constant

**Review by Sofiya-kumar — CHANGES_REQUESTED** [review]

File: Trimble.Modus.Components/Controls/Button/CustomButton.xaml.cs [hunk 5]
```diff
@@ -110,174 +122,235 @@ public new Color BackgroundColor
...
+                break;
+            case ButtonStyle.BorderLess:
+
+                    return (Color)ResourcesDictionary.ColorsDictionary(ColorsConstant.BluePale);
+
+            case ButtonStyle.Fill:
+                {
+                    switch (ButtonColor)
+                    {
+                        case ButtonColor.Primary:
```
Comment (Sofiya-kumar) [inline Trimble.Modus.Components/Controls/Button/CustomButton.xaml.cs:310]: Create a method for this switch case

**Review by Sofiya-kumar — CHANGES_REQUESTED** [review]

**Review by rthanga1 — APPROVED** [review]

**Pull request merged**